### DISCUSSION TITLE
Update Firebase authentication example to include SSR token refresh

### DISCUSSION THREAD

**Discussion by kmjennison** [vercel/next.js#14783] [Help]
# Feature request

## Is your feature request related to a problem? Please describe.

See PR and discussion here: #15202 

The [existing Firebase auth example](https://github.com/vercel/next.js/tree/canary/examples/with-firebase-authentication) sets a cookie with the Firebase user info, which is then available for server-side rendering. However, Firebase tokens expire after 1 hour, which means the server side won't have access to the user when they revisit the site an hour after their last visit, because the token and associated cookie have expired.

This won't work for apps that always need to SSR and need the Firebase ID token to SSR. These apps need additional logic to refresh the Firebase ID token on the server side. We should add this to the Firebase auth example.

## Describe the solution you'd like

There's a PR here with a working server-side token refresh: #15202 

We should merge its logic with the [with-firebase-authentication](https://github.com/vercel/next.js/tree/canary/examples/with-firebase-authentication) to make it easier for users to choose SSR, as suggested in [this review comment](https://github.com/vercel/next.js/pull/15202#issuecomment-673264989).

## Describe alternatives you've considered

N/A

## Additional context

N/A

**Comment by thanhtutzaw** [comment]
Hi This is what is happening in my SSR app . I have used your library next-firebase-auth . But Now I am not using this in current app. 
I get token expire message . Is there anything can you guide me to learn how to solve this issue . Thanks.
https://www.reddit.com/r/Firebase/comments/18r8owf/firebase_id_token_has_expired_get_a_fresh_id/
